public/components/register-agents.js

```diff
@@ -0,0 +1,290 @@
+import React, { Component, Fragment } from 'react';
+
+import { EuiSteps, EuiFlexGroup, EuiFlexItem, EuiPanel, EuiButtonToggle, EuiFieldText, EuiText, EuiCodeBlock, EuiTitle, EuiButtonIcon, EuiButtonEmpty, EuiCopy } from '@elastic/eui';
+
+export class WazuhRegisterAgents extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      status: 'incomplete',
+      selectedOS: '',
+      serverAddress: '',
+    };
+  }
+
+  selectOS(os) {
+    this.setState({ selectedOS: os })
+  }
+
+  setServerAddress = e => {
+    this.setState({ serverAddress: e.target.value })
+  }
+
+  render() {
+    const rpmButton = (
+      <EuiButtonToggle
+        label='Red Hat / CentOS'
+        onChange={(e) => this.selectOS('rpm')}
+        fill={this.state.selectedOS === 'rpm'}
+      />
+    );
+    const debButton = (
+      <EuiButtonToggle
+        label='Debian / Ubuntu'
+        onChange={(e) => this.selectOS('deb')}
+        fill={this.state.selectedOS === 'deb'}
+      />
+    );
+    const windowsButton = (
+      <EuiButtonToggle
+        label='Windows'
+        onChange={(e) => this.selectOS('win')}
+        fill={this.state.selectedOS === 'win'}
+      />
+    );
+    const macOSButton = (
+      <EuiButtonToggle
+        label='MacOS'
+        onChange={(e) => this.selectOS('macos')}
+        fill={this.state.selectedOS === 'macos'}
+      />
+    );
+
+    const ipInput = (
+      <EuiFieldText
+        placeholder="Server address..."
+        value={this.state.serverAddress}
+        onChange={this.setServerAddress}
+      />
+    );
+
+    const copyButton = {
+      position: 'relative',
+      float: 'right',
+      zIndex: '1000',
+      right: '8px',
+      top: '16px'
+    };
+
+    const codeBlock = {
+      zIndex: '100',
+    };
+
+    const rpmText = `rpm --import http://packages.wazuh.com/key/GPG-KEY-WAZUH\ncat > /etc/yum.repos.d/wazuh.repo <<\EOF\n[wazuh_repo]\ngpgcheck=1\ngpgkey=https://packages.wazuh.com/key/GPG-KEY-WAZUH\nenabled=1\nname=Wazuh repository\nbaseurl=https://packages.wazuh.com/3.x/yum/\nprotect=1\nEOF`;
```

Comment:
Move all these declarations to a file named `public/controllers/agent/components/register-guide-defs.js`

There you should expose an object with all the key/value pairs.